xbmc/interfaces/builtins/ProfileBuiltins.cpp

```diff
@@ -66,12 +67,20 @@ static int MasterMode(const std::vector<std::string>& params)
   {
     g_passwordManager.bMasterUser = false;
     g_passwordManager.LockSources(true);
+
+    // master mode turned off, refresh favourites due to possible visibility changes
+    CServiceBroker::GetFavouritesService().RefreshFavourites();
+
     CGUIDialogKaiToast::QueueNotification(CGUIDialogKaiToast::Warning, g_localizeStrings.Get(20052),g_localizeStrings.Get(20053));
   }
   else if (g_passwordManager.IsMasterLockUnlocked(true))
   {
     g_passwordManager.LockSources(false);
     g_passwordManager.bMasterUser = true;
+
+    // successfully entered master mode, refresh favourites
+    CServiceBroker::GetFavouritesService().RefreshFavourites();
```

Comment:
The comment should contain why this needs to be called here. Not a general reason but the specific reason.